DragaliaAPI/Features/Event/RaidEventController.cs

```diff
@@ -0,0 +1,84 @@
+﻿using DragaliaAPI.Controllers;
+using DragaliaAPI.Features.Reward;
+using DragaliaAPI.Features.Trade;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services;
+using DragaliaAPI.Shared.MasterAsset;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DragaliaAPI.Features.Event;
+
+[Route("raid_event")]
+public class RaidEventController(
+    IUpdateDataService updateDataService,
+    IRewardService rewardService,
+    IEventService eventService,
+    ITradeService tradeService
+) : DragaliaControllerBase
+{
+    // TODO: Friendship, passive boosts, summons
+
+    [HttpPost("get_event_data")]
+    public async Task<DragaliaResult> GetEventData(RaidEventGetEventDataRequest request)
+    {
+        RaidEventGetEventDataData resp = new();
+
+        resp.is_receive_event_damage_reward = await eventService.GetCustomEventFlag(
+            request.raid_event_id
+        );
+        resp.raid_event_user_data = await eventService.GetRaidEventUserData(request.raid_event_id);
+        resp.raid_event_reward_list = await eventService.GetEventRewardList<RaidEventRewardList>(
+            request.raid_event_id
+        );
+        resp.event_passive_list = new List<EventPassiveList>
+        {
+            await eventService.GetEventPassiveList(request.raid_event_id)
+        };
+
+        if (
+            MasterAsset.EventTradeGroup.Enumerable
+                .FirstOrDefault(x => x.EventId == request.raid_event_id)
+                ?.Id is
+            { } tradeGroupId
```

Comment:
why do you do this over ?.Id == tradeGroupId?

Reply:
because this actually checks for null and returns the casted/null-checked value as tradeGroupId, the request doesnt actually provide that